frontend/src/components/desktop_apps/AppShortcut.tsx

```diff
@@ -7,43 +7,58 @@ interface AppShortcutProps {
   setOpen: (at: AppType) => void;
   icon?: string;
   useSmaller?: boolean;
+  visible?: boolean;
 }
 
-export default function AppShortcut({ appType, setOpen, icon, useSmaller = false }: AppShortcutProps) {
+export default function AppShortcut({ appType, setOpen, icon, useSmaller = false, visible = true }: AppShortcutProps) {
   const appTextColor: string = useSelector((state: IGameState) => state.styleData.textColor.app);
   const appTextFont: string = useSelector((state: IGameState) => state.styleData.textFont.app);
-
   const shownIcon = `url(assets/app_icons/${icon ?? `${appType.toLowerCase()}`}.png)`;
   const size = useSmaller ? "64px" : "120px";
-  const appTextSize = useSmaller ? "18px" : "20px";
+  const appTextSize = setSize(useSmaller);
 
-  return (
-    <div
-      style={{
-        color: appTextColor,
-        textAlign: "center",
-        fontSize: appTextSize,
-        fontFamily: appTextFont
-      }}
-    >
+  if (visible) {
+    return (
       <div
         style={{
           color: appTextColor,
-          background: shownIcon,
-          width: size,
-          height: size,
-          backgroundSize: "cover",
-          marginLeft: "auto",
-          marginRight: "auto",
-          imageRendering: "pixelated",
+          textAlign: "center",
           fontSize: appTextSize,
-          flexWrap: "wrap"
-        }}
-        onClick={() => {
-          setOpen(appType);
+          fontFamily: appTextFont
         }}
-      ></div>
-      {appType}
-    </div>
-  );
+      >
+        <div
+          style={{
+            color: appTextColor,
+            background: shownIcon,
+            width: size,
+            height: size,
+            backgroundSize: "cover",
+            marginLeft: "auto",
+            marginRight: "auto",
+            imageRendering: "pixelated",
+            flexWrap: "wrap",
+            backgroundRepeat: "no-repeat"
+          }}
+          onClick={() => {
+            setOpen(appType);
+          }}
+        ></div>
+        {appType}
+      </div>
+    );
+  }
+
+  return;
+}
+function setSize(useSmaller: boolean) {
```

Comment:
`useSelector()`'s value should be stored in a variable and then checked for `null` in the if instead of calling it twice.